### DISCUSSION TITLE
Need help with users creation and authentication

### DISCUSSION THREAD

**Discussion by adrianboangiu** [requarks/wiki#7092] [Help / Questions]
Hello,
I am trying to implement a wiki using Wiki.js and I have currently two environments where I experience the same problems with the users creation. Here are the descriptions of my environments:
1. 
OS: Windows10 and Docker
Wiki.js version: 2.5.301
Database engine: PostgreSQL 15.6
Node.js: 18.19.0

2. 
OS: Windows2019
Wiki.js version: 2.5.301
Database engine: PostgreSQL 16.2
Node.js: 20.11.1

I have configured the Mail successfully without Secure and Verify SSL Certificate and I am using Local Authentication strategy. I am using Let’s Encrypt as SSL provider and the HTTP port 3000.
In the documentation it is stipulated:

> You must setup the mail SMTP settings (under Mail in the Administration Area) for the registration emails to be sent. Users won't be able to login until they verify their account, using a link sent to their email address.

However, there is no registration email sent when a new user is created. Therefore, the accounts are not verified and still the users can login. According to the documentation this should not happen. Is this a bug or there is something I did wrong?

According to the documentation the Two-Factor Authentication

>  is available from version 2.5 and up.

However, the Two Factor Authentication (2FA) is not working in my environments. When it is activated on any user and that user do login it receives “Incorrect TFA Security Code” error when it verifies the security code generated by Google or Microsoft Authenticator for the QR Code and the 2FA gets inactive on the user.
Again, is this normal or there is something I missed in the installation or setup procedure?

Thank you

**Comment by triip0d** [comment]
Hey Adrian,

Were you able to solve this issue? I am having the same exact problem with emails not being sent when creating a new user, and I can't seem to find a solution to this.

Thank you

**Comment by Xmoo26** [comment]
The problem has to do with timing. I tried the following: scanned to QR-code into Duo-Mobile. Tried the response code and I would get "Incorrect TFA Security Code". When the minute has passed, it suddenly works. Don't know why. Or maybe it works after 2/3 retries. But eventually the 2FA does work.

Regarding the "However, there is no registration email sent when a new user is created." It is still marked on their to-do list. After all these years, this option is net yet available.

**Comment by justinbornais** [comment]
I am also facing the same problem here. My version is above 2.5, and it won't work (using Microsoft Authenticator). Waiting a minute or two doesn't work, neither does trying multiple times.
